File: 0338-counting-bits/0338-counting-bits.java
class Solution {
    public int[] countBits(int n) {
        int[] ans = new int[n+1];
        for(int i=0; i<=n; i++){
            int sum = 0, num = i;
            while(num>0){
                sum += num%2;
                num = num/2;
            }
            ans[i] = sum;
        }
        return ans;
    }
}

File: 0338-counting-bits/0338-counting-bits.py
class Solution:
    def countBits(self, n: int) -> List[int]:
        ans = [] * (n+1)
        for i in range(n+1):
            cnt, num = 0, i
            while num > 0:
                cnt += num%2
                num = num//2
            ans.append(cnt)
        return ans